src/_hmmc.cpp

```diff
@@ -26,7 +27,33 @@ double logsumexp(double const* v, ssize_t n)
   return std::log(acc) + max;
 }
 
-std::pair<py::array_t<double>, py::array_t<double>> forward_scaling(
+void log_mask_zero_1d(py::array_t<double> probs_,
+  py::array_t<double> log_probs_){
+  auto probs = probs_.unchecked<1>();
+  auto nc = probs.shape(0);
+  auto log_probs = log_probs_.mutable_unchecked<1>();
+
+  for(auto i = 0; i < nc; ++i){
+     log_probs(i) = std::log(probs(i));
+  }
+
```

Comment:
I guess that to be perfectly clean you need to clear the C-level floating point exception flag here and below (if any has been set by a zero probs).  You can probably just inline the code at https://github.com/numpy/numpy/blob/54c2c1ebd290c2cc2faa6f6272f4b04fd0c1eea5/numpy/core/src/npymath/ieee754.c.src#L619

Reply:
Is it enough to test/clear `FE_DIVBYZERO` or do we want to do all of them that numpy did?

Reply:
Let's just clear it for now and open an issue that we may want to copy numpy's entire thing in the future.

Reply:
done.